### DISCUSSION TITLE
Roadmap for making jscad.app official on openjscad.xyz

### DISCUSSION THREAD

**Discussion by hrgdavor** [hrgdavor/jscadui#119] [Ideas]
This is a start of a roadmap to ironing out things on jscad.app to make an official versioned build.

# Missing features
Missing features that are in openjscad.xyz and not in jscad.app
 - [x] auto zoom (zoom to fit)
 - [ ] auto rotate
 - [ ] splash screen explain drag and drop editing
 - [ ] button to show splash screen in menu
 - [ ] splash screen re-appear after few months if hidden
 - [ ] button to open a project
 - [ ] language selector
 - [ ] pick other themes, aside from light/dark
 - [ ] worker timeout (or a button to kill worker if taking too long)
 - [ ] keyboard commands and ability to change them
 - [ ] warning when leaving / refreshing the page

**Comment by z3dev** [comment]
Nice start @hrgdavor Obviously, you have been thinking about this.

I will start playing with the new UI more. Hopefully, @platypii will add a few more ideas.

One thing... I don't think a single website for V1 .vs. V2 .vs. V3 should be created. There are special behaviors / quirks in each version, and users might be more interested in 100% backwards compatible behavior.

I guess we need some users as well. :)

**Comment by z3dev** [comment]
By the way, @platypii should have admin permissions on the openjscad website. So, introduction / documentation / etc pages should be possible. As well as the design pages.

**Comment by z3dev** [comment]
Let's KISS principle the initial website. And go from there.

**Reply by hrgdavor** [reply]
The list is all I could find to be missing. @z3dev which points are you ok with to not be implemented in new version ?

**Comment by platypii** [comment]
Themes: Why would we want themes besides light and dark? No one uses the million random (and mostly ugly) color themes. That was terrible UX.

Warning on leave: I would prefer to NOT warn users when they leave, the popup is annoying. Better idea to just save the current design to localStorage so that if the user returns, their work is not lost?

The rest make sense to me. But to @z3dev's point... why not keep it simple? I don't think jscad.app should have exactly the same features as xyz, I think its better to be simpler.

**Reply by z3dev** [reply]
Yeah. Too many themes. But I do use light/dark/grass. I need a different view of designs sometimes, especially working with 2D.

**Comment by z3dev** [comment]
Button to open a project. It's not necessary if users can figure out to drag and drop files/folders. So, maybe the intro page should have a small introduction.
